src/core/public/execution_context/execution_context_container.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { KibanaExecutionContext } from '../../types';
+
+// Switch to the standard Baggage header
+// https://github.com/elastic/apm-agent-rum-js/issues/1040
+const BAGGAGE_HEADER = 'x-kbn-context';
+
+// Maximum number of bytes per a single name-value pair allowed by w3c spec
+// https://w3c.github.io/baggage/
+export const BAGGAGE_MAX_PER_NAME_VALUE_PAIRS = 4096;
+
+// a single character can use up to 4 bytes
+const MAX_BAGGAGE_LENGTH = BAGGAGE_MAX_PER_NAME_VALUE_PAIRS / 4;
+
+// the trimmed value in the server logs is better than nothing.
+function enforceMaxLength(header: string): string {
+  return header.slice(0, MAX_BAGGAGE_LENGTH);
```

Comment:
An alternative implementation might be to sort all the object values by length and attached them to a header while the header size < 4096. I think we can start with the simplest option.

Reply:
If I'm reading the spec correctly, each list-member can be up to 4096 bytes / 4 = 1024 characters, but the whole header value is limited to 8192 bytes / 4 = 2048 characters. Since we're just enforcing on the whole header value, should we allow up to 2048 characters instead of only 1024?

Reply:
`baggage` header is not meant to be used by `execution context` service exclusively. For example, the APM agent can attach some additional metadata as well.
```json
baggage: {
  kibanaExecutionContext: { ... },
  anotherProperty: {...}
}
```
So I stick to a single property limit in 4096 bytes.

Reply:
> If the header value is a serialized json object, wouldn't truncation cause an invalid object in the end? I see we're try/catching on the server-side when parsing the header, but I wonder if this is good enough?

It will. That's why I added a comment that we still can get value from a logged header even if it's not a valid JSON. as noted above, we can implement an alternative solution to make sure it always sends a valid JSON `An alternative implementation might be to sort all the object values by length and attached them to a header while the header size < 4096 bytes`